x/oracle/keeper/owasm.go

```diff
@@ -152,3 +160,42 @@ func (k Keeper) ResolveRequest(ctx sdk.Context, reqID types.RequestID) {
 		k.ResolveSuccess(ctx, reqID, env.Retdata, output.GasUsed)
 	}
 }
+
+// CollectFee subtract fee from fee payer and send them to treasury
+func (k Keeper) CollectFee(ctx sdk.Context, payer sdk.AccAddress, feeLimit sdk.Coins, askCount uint64, rawRequests []types.RawRequest) (sdk.Coins, error) {
+
+	if err := feeLimit.Validate(); err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidCoins, err.Error())
+	}
+
+	collector := newFeeCollector(k.bankKeeper, feeLimit, payer)
+
+	for _, r := range rawRequests {
+
+		ds, err := k.GetDataSource(ctx, r.DataSourceID)
```

Comment:
Can we use `MustGetDataSource` ?

Reply:
Updated

Reply:
Sorry using `MustGetDataSource` is too dangerous so I decide to change it back